util/top_gen/parser.py

```diff
@@ -75,6 +86,19 @@ def check_pins(pins: list[Pin]) -> list[Pin]:
             raise ValueError("All pins must have unique names.")
         return pins
 
+    @field_validator("blocks")
+    @staticmethod
+    def check_blocks(blocks: list[Block]) -> list[Block]:
+        block_names = {block.name for block in blocks}
+        expected_block_names = {"gpio", "i2c", "spi", "uart"}
```

Comment:
Is there a reason why you are hard-coding this? If we added PWM for example we would have to add it in the toml and also here.

Reply:
And also to the templates and RTL. I think this is a good sanity check to have.

Reply:
Ok, I assume you put this in because you ran into issues, so let's resolve this for now.